design/specs/foreign-key-constraints.md

```diff
@@ -2,75 +2,169 @@
+- This opens a dialog from which the user can set all supported constraint types.
+- The user sees a layout with two panels. On one side, all the existing constraints are listed. On the other side, a form with the constraints configuration is displayed. The form corresponds to the currently active list item.
+- The list panel contains actions at the bottom of the panel. From there, the user can choose to add or delete an item from the list. The user clicks on 'Add' to create a new constraint.
+- The user is presented with a form. From there, they select the type of constraint to be added. The user selects the 'Foreign Key' constraint option from the list.
+- The user selects single or multiple columns they wish to apply the constraint to.
+- The user selects a table to be referenced by these columns.
+- The user selects a column in the reference table from which the column will match values. By default, this is set to the 'Primary Key' column in the referenced table. When changing this field, the user can select only columns with a UNIQUE or PRIMARY KEY constraint.
 
----
+## Scenario 2: User Edits an Existing Foreign Key Constraint
+
+### Scenario 2a: From the column header menu
+
+#### Steps for 2a
+
+- The user hovers over the foreign key reference link below the column name.
+- Information about the constraints applied to the column is displayed in a tooltip.
+- The user clicks on the foreign key reference link below the column name.
+- Constraint settings open for the column.
+
+### Scenario 2b: From table constraint settings
+
+#### Steps for 2b
+
+- A user wants to edit an existing foreign key constraint to change the selected columns or select another referenced table.
+- The user can open a specific foreign key constraint form by clicking on the foreign key link in the column header.
+- If the modified FK constraint causes existing column values to become invalid, the user will receive a warning and be aware of the potential implications of this change, like loss of data or broken views.
+
+## Scenario 3: User Identifies a Column With a Foreign Key Constraint Applied
+
+### Scenario 3a: The foreign key constraint is set to a single column
+
+#### Steps for 3a
+
+- The user opens a table that contains columns with a foreign key constraint applied.
+- The user identifies the columns in two ways:
+  - The user looks at the column header and sees the foreign key indicator. This indicator shows a key icon and the referenced table and column next to it.
+    - The user can click the indicator to display the foreign key settings for the specific column.
+    - The indicator can be hovered over to display a tooltip with information about the foreign key constraints.
+  - The user looks at the column fields and sees that values are added with different styling, a tag-like element with a colored background.
```

Comment:
Are the colors per-foreign-table, or just to denote "foreign key"?  I think it would be cool if I could tell at a glance which table was linked by color (could get too busy though).

Reply:
Based on the loom video, the colours are per-foreign-key-table. I like how the colours are different based on the foreign key table rather than the foreign key column.

Reply:
Based on the loom video, the colours are per-foreign-key-table. I like how the colours are different based on the foreign key table rather than the foreign key column.

Reply:
I see that now.  I also like it.

Reply:
A note about the usage of color will be added to the spec. Once we have a design for views we can discuss implementation details.